quantile-regression/README.md

```diff
@@ -0,0 +1,69 @@
+# Quantile Regression
+
+This discusses how we can include quantile regression in tidymodels. 
+
+The discussion below is organized by package
+
+## parsnip
+
+Users will probably have to specify the quantiles at the specification time (see section on new models). 
+
+### Engines and Modes
+
+The main question is: Should we make new engines available via `set_engine()` or create a new mode of `"quantile regression"`? 
+
+Pros of `set_engine()`: 
+
+* Most (but not all) quantile regression predictions will be for numeric modes. 
+* There is not much difference between quantile models that require a new mode (unlike censored regression). For example, our yardstick functions to compute loss for quantile regression can ingest the list column of predictions (similar to dynamic survival models). 
+* Where do we specify the list of quantiles? It would be suboptimal to add a main option to each function. The users could set it in `set_engine(),`, but that would require more specialized parsing of the return value of that function. 
+
+Cons of `set_engine()`: 
+
+* There could be confusion around the type of model and prediction being made. Suppose that `rpart` could make quantile predictions (it cannot). Would use an engine of “part” result in the regular CART model or one optimized via quantile loss? 
+* Since the list of quintiles would be specified upfront, it would be advantageous to do this with the new mode (`set_mode(“quantgreg,” quantiles = (1:9) / 10)`). 
+
+To test using a new mode, there is a [parsnip branch](https://github.com/tidymodels/parsnip/tree/quantile-mode) that enables that so you can use: 
```

Comment:
I suspect this is the preferred way to do this, because of the cons listed above. Thanks for the testing branch. I'll try reimplementing some of our engines off this branch in the next few days and link back here describing any difficulties or benefits I encounter.